infra/build/functions/build_project.py

```diff
@@ -313,30 +325,37 @@ def get_build_steps(  # pylint: disable=too-many-locals, too-many-statements, to
     project_yaml,
     dockerfile,
     config,
-    additional_env=None):
+    additional_env=None,
+    use_caching=False):
   """Returns build steps for project."""
 
   project = Project(project_name, project_yaml, dockerfile)
-
   if project.disabled:
     logging.info('Project "%s" is disabled.', project.name)
     return []
 
   timestamp = get_datetime_now().strftime('%Y%m%d%H%M')
-  build_steps = build_lib.get_project_image_steps(
-      project.name,
-      project.image,
-      project.fuzzing_language,
-      config=config,
-      architectures=project.architectures,
-      experiment=config.experiment)
+  if use_caching:
+    # Use cached built image.
+    build_steps = []
```

Comment:
So here we make the decision that we won't be building anything in the caching run of a OFG run -- we'll rely on builds done asynchronously (as we went over here: https://github.com/google/oss-fuzz/pull/12675#issuecomment-2451122510) 

Just double confirming that this is the intention here.

Reply:
Yes -- we will rely on our existing infra to build chronos images daily.

Reply:
@oliverchang this does not work out of the box with ccache builds. The problem here is that the existing build step is used to copy in the copied harness from OFG.

We still need to build the docker image of a given project, however, the Dockerfiles will now look something like:

```
FROM us-central1-docker.pkg.dev/oss-fuzz/oss-fuzz-gen/PROJECT-ofg-cached-address
COPY 01.c /target/fuzz/path.c
```

Reply:
I think at least

Reply:
I'm not 100% sure about this. In https://github.com/google/oss-fuzz-gen/pull/692 some harnesses are not overwritten correctly, whereas others are. I'm trying to debug what's going on there.

Reply:
Ah, I think I know why some are succeeding: those that does not have a cached image worked correctly. The other ones did not.

Duckdb works.

However:
```sh
$ docker manifest inspect us-central1-docker.pkg.dev/oss-fuzz/oss-fuzz-gen/duckdb-ofg-cached-address
no such manifest: us-central1-docker.pkg.dev/oss-fuzz/oss-fuzz-gen/duckdb-ofg-cached-address:latest
```

Reply:
`ntpsec` however does not work, but it has a cached image. See for example this harness: https://llm-exp.oss-fuzz.com/Result-reports/ofg-pr/2024-11-06-692-oc-20241106-large-generated-20241106/benchmark/output-ntpsec-_src_ntpsec_tests_libntp_decodenetnum_c/index.html

The generated harnesses does not correlate with what is in the coverage report. This is because we're missing copying the harness in for ccached containers.